lib/spack/spack/util/compression.py

```diff
@@ -282,38 +307,108 @@ def decompressor_for(path, extension=None):
 unrecognized file extension: '%s'"
             % extension
         )
+    return extension
 
-    if re.match(r"\.?zip$", extension) or path.endswith(".zip"):
+
+def decompressor_for_nix(extension):
+    """Returns a function pointer to appropriate decompression
+    algorithm based on extension type and unix specific considerations
+    i.e. a reasonable expectation system utils like gzip, bzip2, and xz are
+    available
+
+    Args:
+        path (str): path of the archive file requiring decompression
+    """
+    if re.match(r"zip", extension):
         return _unzip
 
     if re.match(r"gz", extension):
         return _gunzip
 
-    if re.match(r"bz2", extension):
+    if re.match(r"bz", extension):
```

Comment:
The convention I am familiar with says that a `tbz` is a `tar.bz2` but maybe that's not always the case if you're aware of an instance where its not (or any conventions/idioms/etc). I guess this is a good time to ask, do we want to support `bz`? I have no idea if bunzip/associated modules can handle that compression algo, so it would take some time research, but is certainly doable

Reply:
> I guess this is a good time to ask, do we want to support bz?

IMO no (at least not as part of this PR), although stricter regex matching would help if that gets added in the future.